app/uk/gov/hmrc/gform/auth/AuthService.scala

```diff
@@ -78,32 +79,72 @@ class AuthService(
   private val notAuthorized: AuthResult = AuthBlocked("You are not authorized to access this service")
   private val decoder = Base64.getDecoder
 
-  private def performAWSALBAuth()(implicit hc: HeaderCarrier): AuthResult = {
+  private def performAWSALBAuth(assumedIdentity: Option[Cookie])(implicit hc: HeaderCarrier): AuthResult = {
+    Logger.info("ALB-AUTH: Start authorization...")
+
     val encodedJWT: Option[String] = hc.otherHeaders.collectFirst {
-      case (header, value) if header === "x-amzn-oidc-data" => value
+      case (header, value) if header === "X-Amzn-Oidc-Data" => value
     }
 
+    Logger.info(s"ALB-AUTH: JWT -> [${encodedJWT.getOrElse("No ALB JWT")}]")
+
     encodedJWT.fold(notAuthorized) { jwt =>
       jwt.split("\\.") match {
         case Array(header, payload, signature) =>
           val payloadJson = new String(decoder.decode(payload))
           Try(Json.parse(payloadJson)) match {
             case Success(json) =>
-              json \ "username" match {
-                case JsDefined(JsString(username)) => AuthSuccessful(AWSALBRetrievals(username))
-                case _                             => AuthBlocked("Username does not exist in JWT")
+              Json.fromJson[JwtPayload](json) match {
+                case JsSuccess(jwtPayload, _) => {
+                  assumedIdentity match {
+                    case Some(cookie) if (jwtPayload.iss == appConfig.albAdminIssuerUrl) => {
```

Comment:
```scala
case Some(cookie) if (a == b) => {
  line1
  line2
}
case Some(cookie) if (a != b) => {
  line3
  line4
}
```

Can be rewritten like:
```scala
case Some(cookie) =>
  if (a == b) {
    line1
    line2
  } else {
    line3
    line4
  }
```

This way safe us from potential mistake made in negated expression `(a != b)`.

Reply:
Yes. Changed it.